modAionImpl/src/org/aion/zero/impl/cli/Cli.java

```diff
@@ -161,6 +199,124 @@ public int call(final String[] args, final Cfg cfg) {
+
+                                    CfgAion.setNetwork(args[1]);
+                                    CfgAion.setConfFilePath(path + "/config/" + args[1] + "/config.xml");
+                                    CfgAion.setGenesisFilePath((path + "/config/" + args[1] + "/genesis.json"));
+                                    Keystore.setKeystorePath(path + "/keystore");
+                                    return 2;
+
+                                }
+
+                                // -n [network] -d [directory]
+                                // 4 arguments --> config folder --> config/genesis --> db/log/keystore
+                                else if (isValid(args[3])) {
+
+                                    multiCli = true;
+                                    checkAndCreateDatadir(args[3]);
+                                    String PATH = BASE_PATH  + "/" + args[3];
+
+                                    CfgAion.setNetwork(args[1]);
+                                    CfgAion.setConfFilePath(PATH + "/config/" + args[1] + "/config.xml");
+                                    CfgAion.setGenesisFilePath((PATH + "/config/" + args[1] + "/genesis.json"));
+                                    Keystore.setKeystorePath(PATH + "/keystore");
+
+                                    String[] newArgs = Arrays.copyOfRange(args, 2, args.length);
+                                    call(newArgs, cfg);
+
+                                    if(multiCli) {
+                                        cfg.getDb().setDatabasePath(args[3] + "/database");
+                                        cfg.getLog().setLogPath(args[3] + "/log");
+                                    }
+                                    return 2;
+
+                                } else {
+                                    System.out.println("\nInvalid datadir selected!");
+                                    System.out.println("Please choose valid directory name!\n");
+                                    return 1;
+                                }
+
+                            default:
+                                System.out.println("\nInvalid network selected!\n");
+                                System.out.println("--- Available Networks ---");
+                                System.out.println("    mainnet, conquest");
+                                System.out.println("--------------------------\n");
+                                return 1;
+                        }
+
+                    } else {
+                        System.out.println("\nInvalid network selected!");
+                        System.out.println("--- Available Networks ---");
+                        System.out.println("    mainnet , conquest");
+                        System.out.println("--------------------------\n");
+                        return 1;
+                    }
+
+                // Determines database folder path
+                case "-d":
+                case "--datadir":
```

Comment:
Question: should we be overriding the default configs, or also replacing the existing entries? Personally, I'm not a fan about changing the configuration file via the command line. I _believe_ most people assume that the CLI parameters override config file, but don't change them.

If we have this as default behaviour, some people may be in for a nasty surprise.

Reply:
will remove the cfg.toXML(null) so that it will not override the config file and only apply the changes to that single instance